File: CoreJava/mini-proj-shopping-cart/src/com/techlab/shopping/Product.java
package com.techlab.shopping;

import java.io.Serializable;

public class Product implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = -1489515248866328511L;
	private int prodId;
	private String prodName;
	private int unitPrice;
	private int quantity;
	private int discountPercent;
	private float total;
	

//	constructor

	public Product(int prodId, String prodName, int unitPrice, int discountPercent, int quantity) {
		this.prodId = prodId;
		this.prodName = prodName;
		this.unitPrice = unitPrice;
		this.quantity = quantity;
		this.discountPercent = discountPercent;
		getTotal();
	}

	@Override
	public String toString() {
		return "\nProduct ID = " + prodId + ", Product Name = " + prodName + ", Unit Price =" + unitPrice
				+ ", Quantity = " + quantity + ", Discount Given = " + discountPercent + "%";
	}

	// getters
	public int getProdId() {
		return prodId;
	}

	public String getProdName() {
		return prodName;
	}

	public int getUnitPrice() {
		return unitPrice;
	}

	public int getDiscountPercentage() {
		return discountPercent;
	}

	public int getQuantity() {
		return quantity;
	}

	public float getTotal() {
		float discount = ((quantity * unitPrice) * (discountPercent / 100));
		total = (quantity * unitPrice) - discount;
		return total;
	}

	public static long getSerialversionuid() {
		return serialVersionUID;
	}

	public int getDiscountPercent() {
		return discountPercent;
	}

	public void setQuantity(int quantity) {
		this.quantity = quantity;
	}

}

File: C#Folder/ShoppingCartApp/ShoppingCartApp/Model/Product.cs
using System;

namespace ShoppingCartApp.Model
{
    class Product
    {
        private int _id;
        private string _name;
        private double _price;
        private float _discountPercentage;

        public Product(int id, string name, double price, float discountPercentage)
        {
            _id = id;
            _name = name;
            _price = price;
            _discountPercentage = discountPercentage;
        }

        public int Id { get => _id; }

        public string Name { get => _name; }

        public double Price { get => _price; }

        public float DiscountPercentage { get => _discountPercentage; }

        public double TotalCost { get => Price - (Price * DiscountPercentage / 100); }

        public override string ToString()
        {
            return 
                "\n\t\t\tProduct Id : " + Id + ", Name : " + Name + ", Unit Price : " + Price + ", Discount% given : " + DiscountPercentage + ", Final Price : " + TotalCost;
        }
    }
}
